language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -603,11 +591,53 @@ impl<'mm, 'up> FunctionContext<'mm, 'up> {
             self.llvm_builder.position_at_end(entry_block);
         }
 
+        // If for debugging you need bytecode or local types:
+        /*
+        let stackless_bytecode: &Vec<sbc::Bytecode> = &fn_data.code;
+        dbg!(stackless_bytecode);
+        let fun_idx: usize = self.env.get_def_idx().0.into();
+        let mv_bytecode = &self.env.module_env.get_verified_module().function_defs[fun_idx].code;
+        dbg!(mv_bytecode);
+
+        dbg!(&fn_data.local_types);
+        */
+
         // Declare all the locals as allocas
         {
+            // the key = symbol.0 = offset in the stack of named variables
+            let mut named_vars = std::collections::HashMap::new();
+
+            let func_env = &self.env;
+            let mod_env = &func_env.module_env;
+            let symbol_pool = func_env.symbol_pool();
+
+            for (i, mty) in fn_data.local_types.iter().enumerate() {
+                if mty.is_struct() {
+                    if let Some(s) = mty.get_struct(mod_env.env) {
+                        let struct_env = s.0;
+                        let struct_symbol = struct_env.get_id().symbol();
+                        let struct_pos = i;
+                        named_vars.insert(struct_pos, struct_symbol);
+                        let fields_cnt = struct_env.get_fields().count();
+                        for field in struct_env.get_fields() {
+                            let f_sym = field.get_name();
+                            let f_offset = field.get_offset();
+                            let f_idx = (struct_pos - fields_cnt) + f_offset;
```

Comment:
I found a problem with this computation:
```Rust
let f_idx =(struct_pos - fields_cnt) + f_offset;
```
For structs coming out of Pack it works fine, but for the structs of Unpack it is a reverse order of the fields vs  structure. I need to find out how to distinguish these two sorts of structures in the local fields.

Reply:
The dataflow relationship is more obvious if one examines the stackless bytecode:
```
0: $t1 := move($t0)
1: ($t2, $t3, $t4) := unpack Country::Country($t1)  
2: destroy($t4)
3: destroy($t3)
4: return $t2
```
Here `$tN` are names the printer gives to localN. We can see `$t2` is the target of the `Operation::Unpack` (which eventually produces LLVM `extractvalue` instructions). So the easiest approach to get some names for locals without doing a full-blown dataflow or slicing analysis is to examine:

1. The locals targeted by `Operation::Unpack`.
2. The locals consumed by `Operation::Pack`.
3. The local extracted by `Operation::BorrowField`.

It should be straightforward to then create a map from each local to a corresponding field name of the struct operated on by one of the three cases above.

Reply:
> will it help to enable the names for 'packed' structs for now? That will help merge the patch and subsequent patches can address the unpacked ones.

I'm thinking of more reliable instrument of attributing the fields to the structs. Unfortunately stackless does not care to preserve name definitions for the local Primitives - or at least I cannot find this info. And this all pushes me to a thought that only the Type definitions are available - I can find the names for the fields and for the structs. 
I'm working now on mapping structs in the bytecode to structs in the locals - maybe this will give me a reliable attribution.

Jason is absolutely right - I was trying to follow the structs processed in Pack (and follow indexing copied from here), but then my code started to catch all sort of the structs and this computation fell down.